### ISSUE TITLE
webSocket works with firefox, not with chrome and safari

### ISSUE THREAD

**Issue by chmike** [gobwas/ws#147]
I'm just testing ws with a very basic websocket application. This is the code on the server side:

```
func webSocket(w http.ResponseWriter, r *http.Request) {
	conn, _, _, err := ws.UpgradeHTTP(r, w)
	if err != nil {
		log.Println("error:", err)
		http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
		return
	}
	log.Println("webSocket opened")
	go func() {
		defer conn.Close()

		var i int
		for {
			msg := fmt.Sprintf("message %d", i)
			err = wsutil.WriteServerText(conn, []byte(msg))
			if err != nil {
				log.Println("error:", err)
				break
			}
			time.Sleep(2 * time.Second)
			i++
		}
	}()
}
```

The output I get is 
```
$ go run .
2021/09/05 09:27:50 Listing URL: http://localhost:3000
2021/09/05 09:28:10 webSocket opened
2021/09/05 09:28:18 error: write tcp [::1]:3000->[::1]:55717: write: broken pipe
```

On the client side, I have this code in a vuex plugin:

```
var webSocket = new WebSocket("ws://localhost:3000/status", "protocolOne");

export default function createWebSocketPlugin() {
    return store => {
        webSocket.onmessage = function (event) {
            console.log("webSocket: on message: ", event);
        }
        webSocket.onopen = function (event) {
            console.log("webSocket: on open: ", event)
            store.dispatch('connectionOpened');
        }
        webSocket.onclose = function (event) {
            console.log("webSocket: on close: ", event)
            store.dispatch('connectionClosed');
        }
        webSocket.onrror = function (event) {
            console.log("webSocket: on error: ", event)
        }
    };
}
```
The function `createWebSocketPlugin`is called once at startup. With firefox, I do see the messages printed to the console as expected. With chrome, I have an error message in the console saying that it failed to connect to `"ws://localhost:3000/status"`. On the server side, I see that the upgrade succeeds, but it reports a broken connection.  With safari, it also fails since I have the broken connection error on the server side, but I can't see the console. 

Could it be because chrome and safari don't support unsecure websockets (ws) ?

**Comment by chmike** [comment]
According to [this stackoverflow response](https://stackoverflow.com/a/50861413/75517), chrome does not allow unsecure websockets to localhost. That's not convenient for development. I must find out what certificate I can use for localhost. Any help would be welcome.

**Comment by cristaloleg** [comment]
Hi, check mkcert project.

**Comment by chmike** [comment]
I didn't try mkcert yet. 

I changed my configuration. The http web server with the webSocket is set as a reverse_proxy of a web site with lets encrypt certificate. I manage to make the webSocket connection with firefox on my Ubuntu computer. The messages are displayed in the console and now also in the web page where it is updated at each new message arrival. The configuration is thus working.

Unfortunately, it doesn't work with Chromium, the open source chrome for Ubuntu. With chromium I get a more explicit error message in the console:  `Error during WebSocket handshake: Sent non-empty 'Sec-WebSocket-Protocol' header but no response was received`. 

Looking at [the webSocket specification](https://developer.mozilla.org/en-US/docs/Web/HTTP/Protocol_upgrade_mechanism) it seam that Chromium requests some options which are apparently not supported or ignored by ws. I don't know what options.

**Comment by chmike** [comment]
I tried with gorilla webSocket and had the same behavior. It works with firefox, and doesn't work with chromium.

**Comment by chmike** [comment]
The problem is solved with some help in a gorilla/webSocket issue. 

The problem is that I added `protocolOne` to the socket creation on the front end side. This is given in a confusing example I found here: https://developer.mozilla.org/en-US/docs/Web/API/WebSockets_API/Writing_WebSocket_client_applications.

Also, it seam untrue that chrome would prevent an unsecure connections to a localhost port. I will check when I have time. 

Now changing my code into

```
var webSocket = new WebSocket("ws://localhost:3000/status");
```
fixes the issue.
